### ISSUE TITLE
Aruco appears to no longer be working: could not find aruco_detector

### ISSUE THREAD

**Issue by dcellucci** [twistedfall/opencv-rust#382]
1. Windows
2. Choco, and then i had to manually add and build the opencv_contrib packages.
3. OpenCV 460
4. rust 1.64.0

Does the Aruco library still compile on the current version of rust-opencv (0.70.0) and OpenCV (4.6)? I see in `aruco.rs` `detect_markers` is labeled depreciated, and I get an error when I try to compile:
```
error[E0433]: failed to resolve: could not find `aruco_detector` in the crate root
   --> C:\Users\visua\.cargo\registry\src\github.com-1ecc6299db9ec823\opencv-0.70.0\src\opencv\hub\aruco.rs:293:209
    |
293 | ...ray, parameters: &core::Ptr<crate::aruco_detector::DetectorParameters>, rejected_img_points: &mut dyn core::ToOutputArray) -> Result<(...
    |                                       ^^^^^^^^^^^^^^ could not find `aruco_detector` in the crate root
```
Commenting the two functions that have a parameter that requires `aruco_detector` results in a successful install, but I need those functions. 
I am hoping that I messed something up in the OpenCV install.

**Comment by twistedfall** [comment]
That's weird, `DetectorParameters` should be in the `aruco` module, not `aruco_detector` (which doesn't even exist as a separate module). I would need the output of `cargo build -vv` after doing `cargo clean` to be able to investigate.

**Comment by dcellucci** [comment]
Yes sorry - long day yesterday so I forgot to attach the gist!
Here's that output:
https://gist.github.com/dcellucci/3c6732a135f4a98ec9c86603f08f5b50

Full disclosure: I installed OpenCV using choco, and then realized that I also needed to install the contrib modules to get Aruco to work. So the resulting paths point to an `install` directory that's the result of me following these instructions detailing how to use CMake to manually build opencv: https://stackoverflow.com/questions/37517983/opencv-install-opencv-contrib-on-windows

**Comment by krithik-ranjan** [comment]
Hi, I have been facing this exact issue for the last couple of days. I realized that although the code in the repo uses `crate::aruco::DetectorParameters`, when I build my project with this crate, the `aruco.rs` file gets overwritten by this file https://gist.github.com/krithik-ranjan/776b60fdc86d9815255e366080880266 which might be the cause. I discovered this by cloning the repo and linking it to my crate locally. Using `git diff`, I notice that several other files in `src/opencv/hub` are changed as well. 
I hope this helps debug the issue. Have @twistedfall or @dcellucci found a workaround for the time being?

**Comment by twistedfall** [comment]
Can you please share the OpenCV version and OS you're using?

**Comment by twistedfall** [comment]
As far as I can see `aruco_detector.hpp` (or any traces of "aruco_detector") only exist in the git version of `opencv_contrib`. The 4.6.0 release (or 4.6.0 tag: https://github.com/opencv/opencv_contrib/tree/4.6.0) doesn't have this and should work fine.

**Comment by twistedfall** [comment]
But now it's also fixed in version 0.74.0
